lib/mutant/env/bootstrap.rb

```diff
@@ -95,7 +95,7 @@ def scope_name(scope)
       # @api private
       def infect
         config.includes.each(&$LOAD_PATH.method(:<<))
-        config.requires.each(&method(:require))
+        config.requires.each(&Kernel.method(:require))
```

Comment:
Short: To be able to spec the global side effect this code should trigger.

Long:

When specifying requires, its IMO okay to rely on the fact `Kernel#require` will do the right thing when called. So its IMO fine to place message expectations to "specify" certain requires happen.

I could actually have setup a real file require it and observe a constant was added, but that would defeat the idea of a unit spec. A unit spec under execution should not mutate global state. (OT: Thats a nice way to try defining unit spec vs integration spec!)

Now when setting up message expectations (that are temporal monkey patches effectively) we need an object to target this monkey patches on.

2 thoughts arise from that:

* The primary API of this class is `Bootstrap.call` that returns `Env`. No instance of `Bootstrap` is visible to the API client. The whole intention is to cause side effects and produce a "stable" environment object. Tests should only use the API of the class and do not "fuck around" with their implementation details. As I do not have access on an instance of `Bootstrap` via the API I do not have an object to set the message expectation up against.

* If I'd even have an API where I could reach instances of `Bootstrap` setting up a message expectation on the object under test would be a very bad idea. Here I actually have to blame ruby because very important APIs with global side effects such as `Kernel#require` are available as private methods everywhere. I'd prefer it if `require 'foo'` would only work toplevel, or at class level, but not at instance level per default. This would reduce lots of confusion.

In summary: By changing the code to another construct, causing the same side effect. Its now more testable.